### PR TITLE
Emmett sandbox

### PR DESCRIPTION
**Pull request by Emmett-Byrne**

### CODE REVIEW

File: ARGO/include/LevelManager.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+class LevelManager
+{
+public:
+	LevelManager(SDL_Renderer* t_renderer);
+	void setupLevel();
+	void update(BaseSystem* t_system);
+	void render(SDL_Renderer* t_renderer, RenderSystem* t_system);
+	void setToWall(Entity& t_entity);
+	void setToFloor(Entity& t_entity);
+	void createRoom(glm::vec2 startPosition, int width, int height);
```
Comment (CathalRedmond) [inline ARGO/include/LevelManager.h]: t_

File: ARGO/include/Utilities.h [hunk 2]
```diff
@@ -28,9 +28,16 @@ namespace Utilities
 	const int LEVEL_TILE_WIDTH = 30;
 	const int LEVEL_TILE_HEIGHT = 20;
 	const int TILE_SIZE = 64;
+
 	const int PLAYER_RADIUS = 32;
+	const int PLAYER_FIRE_RATE = 1000 / 10;
```
Comment (CathalRedmond) [inline ARGO/include/Utilities.h]: why not just 100 here

File: ARGO/src/CollisionSystem.cpp [hunk 3]
```diff
@@ -258,6 +287,7 @@ void CollisionSystem::playerBulletToEnemy(Entity* t_playerBullet, Entity* t_enem
 	if (t_enemy->getAllComps().at(COMPONENT_ID::COLLIDER_CIRCLE_ID) && circleToCircleCollision(t_playerBullet, t_enemy))
 	{
 		static_cast<HealthComponent*>(t_playerBullet->getAllComps().at(COMPONENT_ID::HEALTH_ID))->setHealth(0); //kill the bullet
+		static_cast<HealthComponent*>(t_enemy->getAllComps().at(COMPONENT_ID::HEALTH_ID))->reduceHealth(1); //kill the bullet
```
Comment (CathalRedmond) [inline ARGO/src/CollisionSystem.cpp]: bad comment

File: ARGO/src/Game.cpp [hunk 4]
```diff
@@ -1,6 +1,16 @@
 #include "stdafx.h"
 #include "Game.h"
 
+
+bool cleanUpEnemies(const Entity& t_entity)
+{
+	if (!static_cast<HealthComponent*>(t_entity.getComponent(ComponentType::Health))->isAlive())
```
Comment (CathalRedmond) [inline ARGO/src/Game.cpp]: you dont need an if here, you can just return the expression

File: ARGO/src/Game.cpp [hunk 5]
```diff
@@ -68,7 +79,12 @@ Game::Game() :
 		m_textTest2.addComponent(new TransformComponent());
 		m_textTest2.addComponent(new TextComponent("pt-sans.ttf", m_renderer, Utilities::LARGE_FONT, false, "Not Static Text", 255, 255, 0, 123));
 
-		setupLevel();
+		m_levelManager.setupLevel();
+		//magic numbers for creating a sandbox level plz ignore.
```
Comment (CathalRedmond) [inline ARGO/src/Game.cpp:79]: No

File: ARGO/src/Game.cpp [hunk 6]
```diff
@@ -68,7 +79,12 @@ Game::Game() :
...
+		m_levelManager.createRoom(glm::vec2(64, 64), 12, 12);
```
Comment (CathalRedmond) [inline ARGO/src/Game.cpp]: magic numbers

File: ARGO/src/LevelManager.cpp [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+void LevelManager::setTileNeighbours()
+{
+	for (auto& tile : m_levelTiles)
+	{
+		glm::vec2 position = static_cast<TransformComponent*>(tile.getComponent(ComponentType::Transform))->getPos();
+		Neighbours* neighbours = static_cast<TileComponent*>(tile.getComponent(ComponentType::Tile))->getNeighbours();
+
+		//top
```
Comment (CathalRedmond) [inline ARGO/src/LevelManager.cpp]: what is with this comment?

**Review by CathalRedmond — CHANGES_REQUESTED** [review]

File: ARGO/include/Utilities.h (hunk 2 above)
Comment (Violet-Byrne) [reply ARGO/include/Utilities.h]: Milliseconds divided by how many times per seconds you want to fire, makes more sense for changing it in the future

**Pull request merged**